controllers/stocks.js

```diff
@@ -44,10 +44,49 @@ const fetchStocks = async (req, res) => {
  * @param req {Object} - Express request object
  * @param res {Object} - Express response object
  */
+/**
+ * @deprecated
+ * WARNING: This API endpoint is being deprecated and will be removed in future.
+ * Please use the updated API endpoint: `/stocks/:userId` for retrieving user stocks details.
+ *
+ * This API is kept temporarily for backward compatibility.
+ */
 const getSelfStocks = async (req, res) => {
   try {
     const { id: userId } = req.userData;
     const userStocks = await stocks.fetchUserStocks(userId);
+
+    res.set(
+      "X-Deprecation-Warning",
+      "WARNING: This endpoint is being deprecated and will be removed in the future. Please use `/stocks/:userId` route to get the user stocks details."
```

Comment:
@vikasosmium this res header gets set properly, but for some it always returns empty list of stocks from both routes, old one & new one, even though i've added multiple stocks for multiple users...
is it a bug or intended?

Reply:
not a bug, it's a feature, it will only allow you to get your data.